packages/jest-haste-map/src/index.ts

```diff
@@ -216,7 +218,6 @@ class HasteMap extends EventEmitter implements IHasteMap {
   private _cachePath = '';
   private _changeInterval?: ReturnType<typeof setInterval>;
   private readonly _console: Console;
-  private _isWatchmanInstalledPromise: Promise<boolean> | null = null;
```

Comment:
this made the check happens once _per project_, while we now do it once per module scope (i.e. once per worker/thread/process)